pytorch_lightning/loggers/wandb.py

```diff
@@ -215,6 +213,18 @@ def version(self) -> Optional[str]:
 
     @rank_zero_only
     def finalize(self, status: str) -> None:
-        # upload all checkpoints from saving dir
+        # save checkpoints as artifacts
         if self._log_model:
-            wandb.save(os.path.join(self.save_dir, "*.ckpt"))
+            # use run name and ensure it's a valid Artifact name
+            artifact_name = re.sub(r"[^a-zA-Z0-9_\.\-]", "", self.experiment.name)
+            # gather interesting metadata
+            metadata = {
+                k: v
+                for k, v in dict(self.experiment.summary).items()
+                if isinstance(v, numbers.Number) and not k.startswith("_")
+            }
+            # TODO: see if we can also log data from `trainer.checkpoint_callback` (best_model_path, etc) 
+            artifact = wandb.Artifact(name=f"run-{artifact_name}", type="model", metadata=metadata)
+            # TODO: we need access to `trainer.checkpoint_callback.dirpath`
+            artifact.add_dir(trainer.checkpoint_callback.dirpath)
```

Comment:
does that mean you log all checkpoints inside that directory?

Wouldn't it be better to only log the relevant parts (accessible through `trainer.checkpoint_callback.kth_best_model_path`, `trainer.checkpoint_callback.best_model_path`, `trainer.checkpoint_callback.last_model_path` )? 

In the worst case you might be saving hundreds of checkpoints nobody needs.

Reply:
Great idea!